cool_seq_tool/data_sources/feature_overlap.py

```diff
@@ -0,0 +1,180 @@
+        :raises FeatureOverlapError: If unable to find associated GRCh38 chromosome
+        :return: Chromosome. 1..22, X, Y. No 'chr' prefix.
+        """
+        aliases, error_msg = self.seqrepo_access.translate_identifier(
+            identifier, "GRCh38"
+        )
+
+        if error_msg:
+            raise FeatureOverlapError(str(error_msg))
+
+        if not aliases:
+            raise FeatureOverlapError(
+                f"Unable to find GRCh38 aliases for: {identifier}"
+            )
+
+        chr_pattern = r"^GRCh38:(?P<chromosome>X|Y|([1-9]|1[0-9]|2[0-2]))$"
+        for a in aliases:
+            chr_match = re.match(chr_pattern, a)
+            if chr_match:
+                break
+
+        if not chr_match:
+            raise FeatureOverlapError(
+                f"Unable to find GRCh38 chromosome for: {identifier}"
+            )
+
+        chr_groupdict = chr_match.groupdict()
+        return chr_groupdict["chromosome"]
+
+    def get_grch38_cds_overlap(
+        self,
+        start: int,
+        end: int,
+        chromosome: Optional[str] = None,
+        identifier: Optional[str] = None,
+        residue_mode: ResidueMode = ResidueMode.RESIDUE,
+    ) -> Optional[Dict]:
+        """Get feature overlap for GRCh38 genomic data
+
+        :param start: GRCh38 start position
+        :param end: GRCh38 end position
+        :param chromosome: Chromosome. 1..22, X, or Y. If not provided, must provide
+            `identifier`. If both `chromosome` and `identifier` are provided,
+            `chromosome` will be used.
+        :param identifier: Genomic identifier on GRCh38 assembly. If not provided,
+            must identifier `chromosome`. If both `chromosome` and `identifier` are
+            provided, `chromosome` will be used.
+        :param residue_mode: Residue mode for `start` and `end`
+        :raise FeatureOverlapError: If missing required fields
+        :return: Feature overlap dictionary where the key is the gene name and the value
+            is the list of CDS overlap (cds_start, cds_stop, overlap_start,
+            overlap_stop). Will return residue coordinates.
+        """
+        if chromosome:
+            if not re.match(r"^X|Y|([1-9]|1[0-9]|2[0-2])$", chromosome):
+                raise FeatureOverlapError("`chromosome` must be 1, ..., 22, X, or Y")
+        else:
+            if identifier:
+                chromosome = self._get_chr_from_alt_ac(identifier)
+            else:
+                raise FeatureOverlapError(
+                    "Must provide either `chromosome` or `identifier`"
+                )
+
+        # GFF is 1-based, so we need to convert inter-residue to residue
+        # RESIDUE           |   | 1 |   | 2 |   | 3 |   |
+        # INTER_RESIDUE     | 0 |   | 1 |   | 2 |   | 3 |
+        if residue_mode == ResidueMode.INTER_RESIDUE:
+            if start != end:
+                start += 1
+            else:
+                end += 1
```

Comment:
I'm not sure about `start == end`. I was going off of HGVS insertion example: http://normalize.cancervariants.org/variation/translate_from?variation=ENST00000331728.9:c.2049_2050insA

Reply:
@ahwagner Can you help with this too? 😅

Reply:
@ahwagner yes

Reply:
Well, maybe not for this method (tbd). But I’d like still like to know the answer

Reply:
I ended up modifying the GFF dataframe by substracting one for the start positions: 
https://github.com/GenomicMedLab/cool-seq-tool/blob/7e4272828d6f1bc962847a6b3aa22a5d645f4385/cool_seq_tool/data_sources/feature_overlap.py#L75-L76

Then modifying user's residue input:
https://github.com/GenomicMedLab/cool-seq-tool/blob/7e4272828d6f1bc962847a6b3aa22a5d645f4385/cool_seq_tool/data_sources/feature_overlap.py#L159-L164

Reply:
Not sure about lines 161-162. How does one specify a 1nt feature?

Reply:
@ahwagner Idk I was just going with if someone passed start=1 and end=1. Is that not something that could happen? If not, I can raise an exception if start == stop.

Reply:
I am just wondering why we handle the case of `start == end` explicitly. In residue coordinates, this represents a 1nt search, which is totally possible. But if we then do `end -= 1`, it is now a 0 nt search, which may not be what was intended. So trying to understand the requirements for the `start == end` block.

Reply:
I was thinking of 0-based, which needs to be fixed in a separate issue.